python_modules/dagster/dagster/_core/definitions/asset_daemon_cursor.py

```diff
@@ -97,281 +91,220 @@ def get_previous_requested_or_discarded_subset(
-    ) -> AssetConditionCursor:
-        partitions_def = asset_graph.get_partitions_def(asset_key)
-        handled_partitions_subset = self.handled_root_partitions_by_asset_key.get(asset_key)
-        if handled_partitions_subset is not None:
-            handled_subset = AssetSubset(asset_key=asset_key, value=handled_partitions_subset)
-        elif asset_key in self.handled_root_asset_keys:
-            handled_subset = AssetSubset(asset_key=asset_key, value=True)
-        else:
-            handled_subset = AssetSubset.empty(asset_key, partitions_def)
-
-        previous_evaluation = self.latest_evaluation_by_asset_key.get(asset_key)
-        return AssetConditionCursor(
-            asset_key=asset_key,
-            previous_evaluation=previous_evaluation,
-            previous_max_storage_id=self.latest_storage_id,
-            previous_evaluation_timestamp=self.latest_evaluation_timestamp,
-            extras=[
-                AssetConditionCursorExtras(
-                    condition_snapshot=_get_placeholder_missing_condition().snapshot,
-                    extras={"handled_subset": handled_subset},
-                )
-            ],
+
+    @staticmethod
+    def empty(evaluation_id: int = 0) -> "AssetDaemonCursor":
+        return AssetDaemonCursor(
+            evaluation_id=evaluation_id,
+            asset_cursors=[],
+            last_observe_request_timestamp_by_asset_key={},
         )
 
-    def with_updates(
-        self,
-        latest_storage_id: Optional[int],
-        evaluation_id: int,
-        newly_observe_requested_asset_keys: Sequence[AssetKey],
-        observe_request_timestamp: float,
-        evaluations: Sequence["AssetConditionEvaluation"],
-        evaluation_time: datetime.datetime,
-        asset_cursors: Sequence[AssetConditionCursor],
+    @staticmethod
+    def from_serialized(
+        raw_cursor: Optional[str], asset_graph: Optional[AssetGraph], default_evaluation_id: int = 0
     ) -> "AssetDaemonCursor":
-        """Returns a cursor that represents this cursor plus the updates that have happened within the
-        tick.
+        """Deserializes an AssetDaemonCursor from a string. Provides a backcompat layer for the old
+        manually-serialized cursor format.
         """
-        result_last_observe_request_timestamp_by_asset_key = {
-            **self.last_observe_request_timestamp_by_asset_key
-        }
-        for asset_key in newly_observe_requested_asset_keys:
-            result_last_observe_request_timestamp_by_asset_key[
-                asset_key
-            ] = observe_request_timestamp
-
-        if latest_storage_id and self.latest_storage_id:
-            check.invariant(
-                latest_storage_id >= self.latest_storage_id,
-                "Latest storage ID should be >= previous latest storage ID",
+        if raw_cursor is None:
+            return AssetDaemonCursor.empty(default_evaluation_id)
+        try:
+            decoded = base64.b64decode(raw_cursor)
```

Comment:
I think it's probably not strictly necessary to do the gzipping (there aren't going to be that many total cursors floating around out there, as we only keep track of the most recent one), but it does make the cursor around 20x smaller. Probably worth discussing in standup tomorrow.

Reply:
Removed the gzipping bit -- this can be re-evaluated in the future but it seems like the db can handle compressing these for us, and again there won't be that many cursors total in the world anyway.